dulwich/client.py

```diff
@@ -502,7 +502,10 @@ def fetch(self, path, target, determine_wants=None, progress=None, depth=None):
 
         """
         if determine_wants is None:
-            determine_wants = target.object_store.determine_wants_all
+            if depth is not None and depth > 1:
+                determine_wants = target.object_store.determine_wants_force
```

Comment:
Do we need to call determine_wants at all in this case, if we're going to want all refs?